### ISSUE TITLE
CPU monitor not monitoring children properly

### ISSUE THREAD

**Issue by graeme-a-stewart** [HSF/prmon#8]
I did a few tests of `prmon` against different numbers of threads and child processes. From what I can see the multi-thread monitoring is in good shape, measuring 76s CPU time from 20s wall x 4 threads. However, the multi-process monitoring is broken, measuring only one process (it seems).

```sh
[root@39bbb7746bc5 prmon]# ./package/tests/burner --time 20 --procs 4 &
[1] 2750
[root@39bbb7746bc5 prmon]# Will run for 20s using 4 process(es) and 1 thread(s)
[root@39bbb7746bc5 prmon]# ./package/prmon --pid $!
[root@39bbb7746bc5 prmon]# cat prmon.json 
{"Max":{"maxVMEM":110752,"maxPSS":2525,"maxRSS":10800,"maxSwap":0,"totRCHAR":6944,"totWCHAR":53,"totRBYTES":0,"totWBYTES":0,"totUTIME":19.100000381469728,"totSTIME":0.29999998211860659,"totCUTIME":0.0,"totCSTIME":0.0},"Avg":{"avgVMEM":110752,"avgPSS":2525,"avgRSS":10800,"avgSwap":0,"rateRCHAR":385,"rateWCHAR":2,"rateRBYTES":0,"rateWBYTES":0}}
```

CPU time about 19.4s.

`top` confirms that `burner` started 4 processes, as expected, so the issue is on the `prmon` side.

**Comment by graeme-a-stewart** [comment]
Ah, looks like the issue is L109-112. The array is set each time to the values found for 1 process, instead of being accumulated for all the processes in the loop over `cpids`.

Aside - the indentation in that function seems awful. We should run it through `clang-format` or something.

**Comment by amete** [comment]
Hi @graeme-a-stewart,

That's an oversight on my part, I just fixed it in a new branch:

https://github.com/HSF/prmon/commits/master-cpu-monitor-multiprocess-bugfix

As for the formatting I actually see it nicely formatted, which is also the case on GitHub website, but perhaps some configuration on my editor part (perhaps mixing tabs w/ spaces or something). I can play around w/ `clang-format`.

**Comment by graeme-a-stewart** [comment]
Fixed in #9
